core/src/main/java/hudson/model/Slave.java

```diff
@@ -308,7 +308,7 @@ public void setNodeProperties(List<? extends NodeProperty<?>> properties) throws
     }
 
     public RetentionStrategy getRetentionStrategy() {
-        return retentionStrategy == null ? RetentionStrategy.Always.INSTANCE : retentionStrategy;
+        return retentionStrategy == null ? RetentionStrategy.INSTANCE : retentionStrategy;
```

Comment:
Due to the bad name of the constant, this is more confusing now.

Reply:
Good point. I'll revert this change.

Reply:
Another option could be to change `RetentionStategy`, deprecating `INSTANCE` and either replacing all its callers with `new RetentionStategy.Always()`, or a new `ALWAYS` constant whose name is not ambiguous.